Added lua .insert()

pascualcam opened the pull request:
<!--
👋 Hi, thanks for sending a PR to Codecademy Docs! 💖
Please fill out all fields below to ensure your PR is reviewed quickly.

**IMPORTANT**

If you would like to receive credit for your contribution to an entry, make sure to have your Codecademy user profile linked to your GitHub account:

1. Go to your Codecademy dashboard.
2. Click on your profile image in the top-right and then choose "Profile".
3. Then click "Edit Profile".
4. In the "GitHub Username" field, add your username (without the @).
5. Click "Save Changes"

Of course, you can opt not to do this and be listed as an "Anonymous contributor", instead. :)
-->

### Description

Added Lua .insert() function metadata.

### Type of Change

- Editing an existing entry (fixing a typo, bug, issues, etc)

### Checklist

<!-- Please check ALL the boxes: -->

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [X] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

<!--
Having trouble with the PR checker? Here are some common issues and resolutions:

- verify_formatting is failing
  - run `yarn format path/to/markdown/file.md` or `yarn format:all` and commit the results
- verify_lint is failing
  - same as above
  - if verify_lint is still failing, running `yarn lint` locally should let you know what needs to be changed by hand
- test is failing
  - ensure any new markdown files have a `Title` and `Description` defined in their metadata
  - ensure any new markdown files only contain alphanumerics and dashes in their file names and have the same name as their parent directory
  - if that looks ok, running `yarn test` locally should let you know what the issue is
-->

SSwiniarski commented:
@pascualcam There are a couple of things you need to correct in this entry before it gets reviewed.

1) This is in the wrong location. The file path for the .md file as specified by the issue should be `docs/content/lua/concepts/tables/terms/insert/insert.md`
2) You are missing a metadata section for this entry. Please review the instructions for entering metadata from the style guide [here](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md#metadata).

Tag me when these issues are addressed and I'll conduct a fist review.

pascualcam commented:
@SSwiniarski I added the metadata as shown in the sample and added the .md file to the appropriate folder. Cannot delete the previous entry that went into the /documentation directory.

SSwiniarski commented:
@pascualcam the file needs to be located at `content/lua/concepts/tables/terms/insert/insert.md` (note that it is in a folder named insert under `/terms`)

Also you need to remove the extra blank lines in your metadata.

Why can you not delete the file in the wrong place? How are you editing this PR?

pascualcam commented:
@SSwiniarski I believe I have fixed both of your comments.

SSwiniarski commented on content/lua/concepts/tables/terms/insert.md:
This file needs to be moved to `content/lua/concepts/tables/terms/insert/insert.md`

pascualcam replied:
Done

SSwiniarski commented on documentation/insert.md:
This file needs to be deleted.

pascualcam replied:
Removed

SSwiniarski commented on content/lua/concepts/tables/terms/insert.md:
```diff
@@ -0,0 +1,45 @@
+  - 'paths/computer-science'
+---
+
+The **`.insert()`** method inserts a value in a table.
```
```suggestion
The **`.insert()`** method inserts a value in a table at a specified index.
```

pascualcam replied:
Implemented suggested updates

SSwiniarski commented on content/lua/concepts/tables/terms/insert.md:
```diff
@@ -0,0 +1,45 @@
+
+## Syntax
+
+The `value` can be inserted at a specific `index`.  
+
```
```suggestion
```
Delete this line.

pascualcam replied:
Implemented suggested updates

SSwiniarski commented on content/lua/concepts/tables/terms/insert.md:
````diff
@@ -0,0 +1,45 @@
+table.insert(table, index, value)
+```
+
+If `index` is not provided, `value` is inserted at the end of `table`. It returns `table`. Stacks, queues, and double queues can be implemented in this way.    
````
```suggestion
The `value` is inserted at `index`. If `index` is not provided, `value` is inserted at the end of `table`. It returns `table`. Stacks, queues, and double queues can be implemented in this way.    
```

pascualcam replied:
Implemented suggested updates

SSwiniarski commented on content/lua/concepts/tables/terms/insert.md:
```diff
@@ -0,0 +1,45 @@
+
+If `index` is not provided, `value` is inserted at the end of `table`. It returns `table`. Stacks, queues, and double queues can be implemented in this way.    
+
+## Codebyte Example
```
```suggestion
## Example
```
We don't have codebytes implemented for lua.

pascualcam replied:
Implemented suggested updates

SSwiniarski reviewed (changes requested):
@pascualcam I have a few review comments. You still need to move the file and delete the extra one. Are you having trouble doing this?

pascualcam commented:
Let me know if you have further comments. Thanks for your feedback and patience while I become more acquainted with git.

SSwiniarski reviewed (approved):
@pascualcam I deleted the extra files since you seemed to have difficulty with it. I also made a few other small edits. Now it looks good for a second review.

caupolicandiaz commented on content/lua/concepts/tables/terms/insert/insert.md:
````diff
@@ -0,0 +1,43 @@
+
+## Syntax
+
+```lua
````
```suggestion
```pseudo
```

caupolicandiaz commented on content/lua/concepts/tables/terms/insert/insert.md:
````diff
@@ -0,0 +1,43 @@
+## Syntax
+
+```lua
+table.insert(table, index, value)
````
```suggestion
table.insert(sometable, index, value)
```

caupolicandiaz commented on content/lua/concepts/tables/terms/insert/insert.md:
````diff
@@ -0,0 +1,43 @@
+table.insert(table, index, value)
+```
+
+The `value` is inserted at `index`. If `index` is not provided, `value` is inserted at the end of `table`. It returns `table`. Stacks, queues, and double queues can be implemented in this way.
````
```suggestion
The `value` is inserted at `index`. If `index` is not provided, `value` is inserted at the end of `sometable`. The function returns the modified, original table.

 Stacks, queues, and double queues can be implemented in this way.
```

caupolicandiaz commented on content/lua/concepts/tables/terms/insert/insert.md:
````diff
@@ -0,0 +1,43 @@
+
+The output would be:
+
+```
````
```suggestion
```shell
```

caupolicandiaz commented on content/lua/concepts/tables/terms/insert/insert.md:
````diff
@@ -0,0 +1,43 @@
+end
+```
+
+The output would be:
````
```suggestion
The code above results in the following output:
```

caupolicandiaz reviewed (changes requested):
@pascualcam I've completed my initial review. Please update the entry, and tag me when you've finished making changes. Thanks

pascualcam commented:
@caupolicandiaz I have implemented your suggested edits

caupolicandiaz reviewed (approved):
@pascualcam @SSwiniarski Looks good to merge
